godot-macros/src/util.rs

```diff
@@ -22,6 +22,13 @@ pub fn strlit(s: &str) -> Literal {
     Literal::string(s)
 }
 
+pub fn bail_error<T>(msg: impl AsRef<str>, tokens: T) -> Error
+where
+    T: Spanned,
+{
+    Error::new_at_span(tokens.__span(), msg.as_ref())
+}
+
```

Comment:
You seem to always use this method as `Err(bail_error(...))`, so you could as well use the existing `bail(...)` 🙂

Reply:
The problem with the existing `bail()?` method is that it (technically) isn't guaranteed to be an error.
So things like:
```
let val;
if condition {
  val = something;
} else {
  bail()?;
}
```
fail to compile because `val` can't be guaranteed to be initialized.
`return Err(bail_error)` instead makes that work because val is initialized in all branches that continue past the if/else.

Reply:
You can use `return bail();` in that case. 

But even if you couldn't, I would not write code for hypothetical scenarios, this `if/else` is not something you have now. In the spirit of YAGNI 😉

Reply:
It's slightly different, but I do have that case. I only wrote it because I ran into an issue. bail()? `require_key_value` can't be guaranteed to return a value.

Though, I can also solve the issue by using `return bail();`, I hadn't realized that it's set up to infer `R` (the Ok type) and so can be returned from anything that returns a `ParseResult`.